ks3-standard/src/commonMain/kotlin/io/ks3/standard/TypeOmittingTransformer.kt

```diff
@@ -0,0 +1,30 @@
+package io.ks3.standard
+
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonTransformingSerializer
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.serializer
+
+/**
+ * A serializer that omits the type discriminator when serializing with polymorphic serialization.
+ *
+ * This is useful when you want to serialize a sealed class hierarchy to JSON without including the type discriminator.
+ *
+ * @param classDiscriminatorProperty The name of the property that is used as the type discriminator. Defaults to "type".
+ */
```

Comment:
Can you say more about the situation where this would be helpful?

I'm a bit cautious about this serializer, because it will create JSON that KxS can't decode, which is a bit weird, and there's no clean way around it.

What about just providing a `SerializationStrategy<Foo>` that must be passed manually?

```kotlin
fun main() {
  val fooWithoutTypeSerializer = object : SerializationStrategy<Foo> {
    override val descriptor: SerialDescriptor
      get() = Foo.serializer().descriptor

    override fun serialize(encoder: Encoder, value: Foo) {
      // ...
    }
  }

  Json.encodeToString(fooWithoutTypeSerializer, Foo.Bar(""))
}
```

That way it's highlights the weirdness of the situation?

Either that or add an opt-in requirement that says "hey, this will create JSON that can be decoded, are you sure?"